File: backend-beeyond/src/main/java/at/htl/beeyond/dto/ApplicationDto.java
package at.htl.beeyond.dto;

import at.htl.beeyond.entity.ApplicationStatus;
import org.hibernate.validator.constraints.Length;

import javax.json.bind.annotation.JsonbTransient;

public abstract class ApplicationDto {

    protected Long id;

    @Length(max = 255)
    protected String note;

    protected ApplicationStatus status;

    protected UserDto owner;

    public ApplicationDto(Long id, String note, ApplicationStatus status, UserDto owner) {
        this.id = id;
        this.note = note;
        this.status = status;
        this.owner = owner;
    }

    public ApplicationDto() {
    }

    public Long getId() {
        return id;
    }

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }

    public ApplicationStatus getStatus() {
        return status;
    }

    @JsonbTransient
    public void setStatus(ApplicationStatus status) {
        this.status = status;
    }

    public UserDto getOwner() {
        return owner;
    }

    @JsonbTransient
    public void setOwner(UserDto owner) {
        this.owner = owner;
    }

}

File: backend-beeyond/src/main/java/at/htl/beeyond/dto/ApplicationDto.kt
package at.htl.beeyond.dto

import at.htl.beeyond.entity.ApplicationStatus
import at.htl.beeyond.entity.Namespace
import at.htl.beeyond.validation.Exists
import java.time.LocalDate
import java.time.LocalDateTime
import java.util.*
import javax.json.bind.annotation.JsonbDateFormat
import javax.json.bind.annotation.JsonbProperty
import javax.json.bind.annotation.JsonbTransient
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

abstract class ApplicationDto(
    var id: Long? = null,
    @field:Size(max = 255) var note: String? = null,
    @set:JsonbTransient var status: ApplicationStatus? = null,
    @set:JsonbTransient var owner: UserDto? = null,
    @set:JsonbTransient var createdAt: LocalDateTime? = null,
    @set:JsonbTransient var startedAt: LocalDateTime? = null,
    @set:JsonbTransient var finishedAt: LocalDateTime? = null,
    @field:NotNull
    @field:Exists(entity = Namespace::class, fieldName = "namespace")
    var namespace: String? = null,
    @field:JsonbProperty("class")
    @field:NotNull var schoolClass: String? = null,
    @set:JsonbDateFormat(value = "dd.MM.yyyy")
    @field:JsonbProperty("to")
    @field:NotNull var toDate: LocalDate? = null,
    @field:Size(max = 255)
    @field:NotNull var purpose: String? = null
){
    abstract var content: String?
}
